src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1106,6 +1107,11 @@ internal static void ValidateName(string name, bool isProxy = false)
             }
         }
 
+        internal static string UpdateProxyName(string name)
+        {
+            return ProxyNameValidationRegex.Replace(name, string.Empty);
```

Comment:
What is this doing - replacing what with what?

Reply:
This is running the regex Regex(@"[^a-zA-Z0-9_-]", RegexOptions.Compiled | RegexOptions.IgnoreCase);, basically removing all non alphanumeric characters and anything other than '_', and '_'

---
In reply to: [146605862](https://github.com/Azure/azure-webjobs-sdk-script/pull/2060#discussion_r146605862) [](ancestors = 146605862)